java/google/registry/rde/XjcToDomainResourceConverter.java

```diff
@@ -0,0 +1,329 @@
+        .setCreationClientId(domain.getCrRr().getValue())
+        .setCreationTime(domain.getCrDate())
+        .setRegistrationExpirationTime(domain.getExDate())
+        .setLastEppUpdateTime(domain.getUpDate())
+        .setLastEppUpdateClientId(domain.getUpRr() == null ? null : domain.getUpRr().getValue())
+        .setLastTransferTime(domain.getTrDate())
+        .setStatusValues(
+            ImmutableSet.copyOf(Iterables.transform(domain.getStatuses(), STATUS_CONVERTER)))
+        .setNameservers(convertNameservers(domain.getNs()))
+        .setGracePeriods(ImmutableSet
+            .copyOf(Iterables.transform(
+                domain.getRgpStatuses(), gracePeriodConverter)))
+        .setContacts(
+            ImmutableSet.copyOf(Iterables.transform(domain.getContacts(), CONTACT_CONVERTER)))
+        .setDsData(
+            domain.getSecDNS() == null ? ImmutableSet.<DelegationSignerData>of() : ImmutableSet
+            .copyOf(Iterables.transform(domain.getSecDNS().getDsDatas(), SECDNS_CONVERTER)))
+        .setTransferData(convertDomainTransferData(domain.getTrnData()));
+    checkArgument(
+        domain.getRegistrant() != null, "Registrant is missing for domain '%s'", domain.getName());
+    builder = builder.setRegistrant(convertRegistrant(domain.getRegistrant()));
+    return builder.build();
+  }
+
+  private static BillingEvent.Recurring createAutoRenewBillingEvent(
+    XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final BillingEvent.Recurring billingEvent = new BillingEvent.Recurring.Builder()
+        .setReason(Reason.RENEW)
+        .setFlags(ImmutableSet.of(Flag.AUTO_RENEW))
+        .setTargetId(domain.getRoid())
+        .setClientId(domain.getClID())
+        .setEventTime(domain.getExDate())
+        .setRecurrenceEndTime(END_OF_TIME)
+        .setParent(historyEntry)
+        .build();
+    ofy().transact(new VoidWork() {
+
+      @Override
+      public void vrun() {
+        ofy().save().entity(billingEvent);
+      }
+    });
+    return billingEvent;
+  }
+
+  private static PollMessage createAutoRenewPollMessage(
+      XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final PollMessage pollMessage = new PollMessage.Autorenew.Builder()
+        .setTargetId(domain.getRoid())
+        .setClientId(domain.getClID())
+        .setEventTime(domain.getExDate())
+        .setMsg("Domain was auto-renewed.")
+        .setParent(historyEntry)
+        .build();
+    ofy().transact(new VoidWork() {
+
+      @Override
+      public void vrun() {
+        ofy().save().entity(pollMessage);
+      }
+    });
+    return pollMessage;
+  }
+
+  private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
+    // TODO: (wolfgang) validate the missing fields in code review
+    XjcRdeDomainElement element = new XjcRdeDomainElement(domain);
+    final HistoryEntry historyEntry = new HistoryEntry.Builder()
+        .setType(HistoryEntry.Type.RDE_IMPORT)
+//        .setPeriod(getCommandPeriod())
```

Comment:
It's worth auditing the existing code to see if any uses of these two require their presence, and if so, we need to have a larger discussion about how to take what are currently required fields and make them optional.  It wouldn't surprise me if some code somewhere was requiring them.  It may also be easier to use sane default values than to make them optional, like always setting period to one year, and generating a new transaction id.  Near as I can tell neither of these are actually used anywhere -- so you could also just generate a new random transaction ID?